### ISSUE TITLE
Preparing for MC generation in 80x

### ISSUE THREAD

**Issue by mulhearn** [cms-l1t-offline/cmssw#331]
This is a super issue to keep track of issues related to production of MC with first pass 2016 L1T trigger emulation. 

Related PRs:
- new repack for MC (80x) cms-sw#14647 (81x) cms-sw#14635 - both merged
- switch to tau option 21 (80x) cms-sw#14646 (81x) cms-sw#14654 - both merged
- add overflow checks to sums (80x) cms-sw#14650 (81x) cms-sw#14651 - both merged

For convenient testing before the above are available in an IB, there is a snowball suitable for use with a recent 80x integration build, like this:

```
cmsrel CMSSW_8_0_X_2016-05-25-2300
cd CMSSW_8_0_X_2016-05-25-2300/src
cmsenv
git cms-merge-topic --unsafe cms-l1t-offline:l1t-809-cand-pass1
```

When using the usual ntuple production, note that we can now drop the `--geometry=Extended2016,Extended2016Reco --customise=L1Trigger/Configuration/customiseReEmul.L1TEventSetupForHF1x1TPs` from all workflows _except_ the re-emulate from 2015 data workflow (which still requires the 2015 conditions to be overriden!).

As a new workflow, we should run the planned MC re-L1T production sequence and save the results into ntuples:

`cmsDriver.py reHLT --conditions auto:run2_mc -s L1REPACK:FullMC,RAW2DIGI --no_output -n 10 --era=Run2_2016 --filein root://eoscms.cern.ch//eos/cms/store/mc/RunIISpring16DR80/TTJets_TuneCUETP8M1_13TeV-amcatnloFXFX-pythia8/RAWAODSIM/PUSpring16RAWAODSIM_80X_mcRun2_asymptotic_2016_v3-v2/30000/040AFBF1-8E0B-E611-9C22-001E67E95A58.root --processName reL1T --python_filename l1t_test.py --customise=L1Trigger/L1TNtuples/customiseL1Ntuple.L1NtupleAODRAWEMU --customise=L1Trigger/Configuration/customiseUtils.L1TStage2DigisSummary --no_exec'

For production, you can remove the `--customise=L1Trigger/Configuration/customiseUtils.L1TStage2DigisSummary` which is included here as quick initial check that the whole chain is working.

This is a super issue to keep track of issues related to production of MC with first pass 2016 L1T trigger emulation.  Pending action items are:

1) Merging l1t-tsg-v7-cand plus increments into 81x and 80x:  #292 (Done)

2) Update GT with L1 menu for MC production:  L1Menu_Collisions2016_v4

3) Update L1REPACK to handle MC (DONE)

Pending validation items are:

4) Rerun our previous tests of bitwise agreement when rerunning MC production over MC raw with latest L1REPACK (3).  We ran similar tests in past and don't expect any problem, but it is possible that the latest emulation depends on something new which breaks, e.g., we have had updates to muon TP emulation.  The bitwise equivalence test config files are generated like this:

`cmsDriver.py L1TEST --conditions=auto:run2_mc -s DIGI,L1,DIGI2RAW --eventcontent RAWSIM --datatier GEN-SIM-RAW -n 5 --era Run2_2016 --mc --no_exec --filein=/store/relval/CMSSW_7_6_0_pre7/RelValTTbar_13/GEN-SIM/76X_mcRun2_asymptotic_v9_realBS-v1/00000/0A812333-427C-E511-A80A-0025905964A2.root --customise_commands 'process.simHcalDigis.markAndPass = cms.bool(True) \n'`

`cmsDriver.py L1TEST --conditions=auto:run2_mc -s RAW2DIGI,L1REPACK:FullMC --no_output -n 5 --era Run2_2016 --mc --no_exec --filein=file:L1TEST_DIGI_L1_DIGI2RAW.root --customise=L1Trigger/Configuration/customiseUtils.L1TStage2ComparisonRAWvsEMU`

RESULT:  Tested at 1000 events level.

5) Check the intended recipe for MC production and confirm that it is compatible with assumptions of (4) and the re-emulation we are using to validate the L1 code.

RESULT:  This looks good, but will scrutinize more in parallel.

**Comment by mulhearn** [comment]
Takashi provides update on item (2).  He has received request for  L1Menu_Collisions2016_v4.xml and it should be available in a few days.  No software changes are expected for L1T code to handle this new version of the menu, it should only be a Global Tag update.

**Comment by mulhearn** [comment]
At last l1t-tsg-v7-cand is in 80x (as well as in 81x) so we are down to the two small "straggler PRs" which will go in once the 81x once (already merged into 81x) pass IB tests.

**Comment by mulhearn** [comment]
This is all merged, and there is a new workflow for @apana and @bortigno to test with.
